File: Editorial/0011_Container_With_Most_Water/Go/2_Two_Pointer_Approach.go
func maxArea(height []int) int {
    maxArea := 0
    left := 0
    right := len(height) - 1

    for left < right {
        width := right - left
        maxArea = max(maxArea, min(height[left], height[right])*width)
        if height[left] <= height[right] {
            left++
        } else {
            right--
        }
    }
    return maxArea
}

File: Editorial/11_Container_With_Most_Water/C/2_Two_Pointer_Approach.c
int max(int a, int b) { return a > b ? a : b; }

int min(int a, int b) { return a < b ? a : b; }

int maxArea(int* height, int heightSize) {
    int maxArea = 0;
    int left = 0;
    int right = heightSize - 1;

    while (left < right) {
        int width = right - left;
        maxArea = max(maxArea, min(height[left], height[right]) * width);
        if (height[left] <= height[right]) {
            left++;
        } else {
            right--;
        }
    }
    return maxArea;
}
